examples/01_plotting/plot_3d_map_to_surface_projection.py

```diff
@@ -136,161 +146,182 @@
 # Use an atlas and choose regions to outline
 # ------------------------------------------
 
-destrieux_atlas = datasets.fetch_atlas_surf_destrieux()
-parcellation = destrieux_atlas["map_right"]
+from nilearn.experimental.surface import fetch_destrieux
+
+destrieux_atlas, label_names = fetch_destrieux(mesh_type="inflated")
 
 # these are the regions we want to outline
 regions_dict = {
-    b"G_postcentral": "Postcentral gyrus",
-    b"G_precentral": "Precentral gyrus",
+    "G_postcentral": "Postcentral gyrus",
+    "G_precentral": "Precentral gyrus",
 }
 
 # get indices in atlas for these labels
 regions_indices = [
-    np.where(np.array(destrieux_atlas["labels"]) == region)[0][0]
-    for region in regions_dict
+    np.where(np.array(label_names) == region)[0][0] for region in regions_dict
 ]
 
 labels = list(regions_dict.values())
 
 # %%
 # Display outlines of the regions of interest on top of a statistical map
 # -----------------------------------------------------------------------
-#
-# Regions can be outlined using both engines.
-# The plotly engine offers more control over the aesthetics of the
-# contours through the lines argument.
-figure = plotting.plot_surf_stat_map(
-    fsaverage.infl_right,
-    texture,
+
+from nilearn.experimental.plotting import plot_surf_contours
+
+fsaverage_sulcal = load_fsaverage_data(data_type="sulcal", mesh_type="pial")
+
+figure = plot_surf_stat_map(
+    stat_map=img,
+    surf_mesh=fsaverage_meshes["inflated"],
     hemi="right",
-    title="Surface right hemisphere",
+    title="ROI outlines on surface",
     colorbar=True,
     threshold=1.0,
-    bg_map=fsaverage.sulc_right,
+    bg_map=fsaverage_sulcal,
     engine=engine,
 )
-
 if engine == "matplotlib":
-    plotting.plot_surf_contours(
-        fsaverage.infl_right,
-        parcellation,
+    plot_surf_contours(
+        roi_map=destrieux_atlas,
+        hemi="right",
         labels=labels,
         levels=regions_indices,
         figure=figure,
         legend=True,
         colors=["g", "k"],
     )
-    plotting.show()
+    show()
 elif engine == "plotly":
     figure.add_contours(
-        roi_map=parcellation,
```

Comment:
This is one function we have not adapted yet to accept SurfaceImage yet
Will open an issue to fix this.